webapp/src/utils.ts

```diff
@@ -77,18 +77,41 @@ export function getChannelURL(state: GlobalState, channel: Channel, teamId: stri
 }
 
 export function getCallsClient(): CallsClient | undefined {
-    return window.opener ? window.opener.callsClient : window.callsClient;
+    let callsClient;
+    try {
+        callsClient = window.opener ? window.opener.callsClient : window.callsClient;
+    } catch (err) {
+        logErr(err);
+    }
+    return callsClient;
+}
+
+export function getCallsClientChannelID(): string {
+    return getCallsClient()?.channelID || '';
+}
+
+export function getCallsClientSessionID(): string {
+    return getCallsClient()?.getSessionID() || '';
+}
+
+export function getCallsClientInitTime(): number {
+    return getCallsClient()?.initTime || 0;
 }
 
 export function shouldRenderCallsIncoming() {
-    const win = window.opener ? window.opener : window;
-    const nonChannels = window.location.pathname.startsWith('/boards') || window.location.pathname.startsWith('/playbooks') || window.location.pathname.includes(`${pluginId}/expanded/`);
-    if (win.desktop && nonChannels) {
+    try {
+        const win = window.opener ? window.opener : window;
+        const nonChannels = window.location.pathname.startsWith('/boards') || window.location.pathname.startsWith('/playbooks') || window.location.pathname.includes(`${pluginId}/expanded/`);
+        if (win.desktop && nonChannels) {
         // don't render when we're in desktop, or in boards or playbooks, or in the expanded view.
         // (can be simplified, but this is clearer)
+            return false;
+        }
+        return true;
+    } catch (err) {
+        logErr(err);
         return false;
```

Comment:
Should we default to returning true like before?

Reply:
Would that make a difference? With the try block in place I am not sure that would be a possible code path, hence why the linter isn't complaining I guess.